gitautosync/__init__.py

```diff
@@ -131,7 +139,33 @@ def _pull_and_resolve_conflicts(self):
 
         yield from execute_cmd(['git', 'checkout', self.branch_name], cwd=self.repo_dir)
         yield from execute_cmd(['git', 'fetch'], cwd=self.repo_dir)
-        yield from execute_cmd(['git', 'merge', '-Xours', 'origin/{}'.format(self.branch_name)], cwd=self.repo_dir)
+
+        try:
+            yield from execute_cmd(['git', 'merge', '-Xours', 'origin/{}'.format(self.branch_name)], cwd=self.repo_dir)
+        except subprocess.CalledProcessError as e:
+            # When a user modifies a file locally, gitautosync first attempts
+            # to commit the change to save the user's work. However, afterwards,
+            # when it tries to merge with upstream, this may cause a merge
+            # conflict if the file is deleted upstream. This is because it is
+            # trying to merge two conflicting changes to the file: one being its
+            # modification and the other being its deletion
+
+            # Because of the merge conflict, git will throw an error ie.
+            #   CONFLICT (modify/delete): new-file.txt deleted in branch1 and
+            #   modified in HEAD. Version HEAD of new-file.txt left in tree.
+            #   Automatic merge failed; fix conflicts and then commit the result.
+
+            # Making a commit and re-merging fixes the conflict by confirming
+            # to git that we want to keep the MODIFIED changes and not the
+            # DELETED changes.
+            # https://stackoverflow.com/questions/4319486/git-merge-conflict-when-local-is-deleted-but-file-exists-in-remote
+
+            # if there was a merge error
+            if e.returncode == 1:
```

Comment:
Returncode can be 1 for a large variety of reasons - I don't think this is specific enough. You might have to read the output to stdout and parse that to verify this is what we're running into.

Reply:
I see. Ok that makes sense. I'll try to parse it then.